sdk/storage/Azure.Storage.Blobs.Batching/src/BlobBatchClient.cs

```diff
@@ -0,0 +1,774 @@
+                try
+                {
+                    if (messages[i].TryGetProperty(BatchConstants.DelayedResponsePropertyName, out object value) &&
+                        value is DelayedResponse response)
+                    {
+                        response.SetLiveResponse(responses[i], throwOnFailure);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    failures.Add(ex);
+                }
+            }
+
+            // Throw any failures
+            if (failures.Count > 0)
+            {
+                throw BatchErrors.ResponseFailures(failures);
+            }
+        }
+        #endregion Create/SubmitBatch
+
+        #region DeleteBlobs
+#pragma warning disable AZC0002 // Client method should have cancellationToken as the last optional parameter
+        /// <summary>
+        /// The DeleteBlobs operation marks the specified blobs for deletion.
+        /// The blobs are later deleted during garbage collection.  All of the
+        /// deletions are sent as a single batched request.
+        /// </summary>
+        /// <param name="blobUris">URIs of the blobs to delete.</param>
+        /// <returns>
+        /// The <see cref="Response"/>s for the individual Delete operations.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure to submit the batch occurs.  Individual sub-operation
+        /// failures will be wrapped in an <see cref="AggregateException"/>.
+        /// </remarks>
+        [ForwardsClientCalls]
+        public virtual Response[] DeleteBlobs(params Uri[] blobUris) =>
+            DeleteBlobsInteral(
+                blobUris,
+                false, // async
+                CancellationToken.None)
+                .EnsureCompleted();
+
+        /// <summary>
+        /// The DeleteBlobs operation marks the specified blobs for deletion.
+        /// The blobs are later deleted during garbage collection.  All of the
+        /// deletions are sent as a single batched request.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be cancelled.
+        /// </param>
+        /// <param name="blobUris">URIs of the blobs to delete.</param>
+        /// <returns>
+        /// The <see cref="Response"/>s for the individual Delete operations.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure to submit the batch occurs.  Individual sub-operation
+        /// failures will be wrapped in an <see cref="AggregateException"/>.
+        /// </remarks>
+        [ForwardsClientCalls]
+        public virtual Response[] DeleteBlobs(
+            CancellationToken cancellationToken,
+            params Uri[] blobUris) =>
```

Comment:
This is an interesting signature do we have any other methods that taka params and cancellation token?

Reply:
No, it's definitely a weirdo.  I think we're going to change this to `DeleteBlobs(IEnumerable<Uri>  blobUris, CancellationToken cancellationToken)`.